pkgs/os-specific/bsd/freebsd/by-name/bmakeMinimal.nix

```diff
@@ -0,0 +1,93 @@
+{ lib, stdenv, mkDerivation, bsdSetupHook, freebsdSetupHook, patchesRoot, hostVersion, ... }:
+mkDerivation {
+  path = "contrib/bmake";
+  extraPaths = [ "share/mk" ]
+    ++ lib.optional (!stdenv.hostPlatform.isFreeBSD) "tools/build/mk";
+
+  buildInputs = [];
+  nativeBuildInputs = [ bsdSetupHook freebsdSetupHook ];
+
+  skipIncludesPhase = true;
+
+  makeFlags = [];
+
+  patches = [
+    /${patchesRoot}/bmake-no-compiler-rt.patch
+  ];
+
+  postPatch = ''
+    patchShebangs configure
+    # make needs this to pick up our sys make files
+    export NIX_CFLAGS_COMPILE+=" -D_PATH_DEFSYSPATH=\"$out/share/mk\""
+
+    # Copied from netbsd. we need to prevent it from trying to adjust the permissions of installed files.
+    #set +e
+    #find $BSDSRCDIR -print0 -name '*.mk' | while IFS= read -r -d "" f; do
+    #  echo $f
+    #  substituteInPlace "$f" \
+    #    --replace '-o ''${DOCOWN}' "" \
+    #    --replace '-g ''${DOCGRP}' "" \
+    #    --replace '-o ''${BINOWN}' "" \
+    #    --replace '-g ''${BINGRP}' "" \
+    #    --replace '-o ''${RUMPBINOWN}' "" \
+    #    --replace '-g ''${RUMPBINGRP}' "" \
+    #    --replace '-o ''${DEBUGOWN}' "" \
+    #    --replace '-g ''${DEBUGGRP}' "" \
+    #    --replace '-o ''${KMODULEOWN}' "" \
+    #    --replace '-g ''${KMODULEGRP}' "" \
+    #    --replace '-o ''${LIBOWN}' "" \
+    #    --replace '-g ''${LIBGRP}' "" \
+    #    --replace '-o ''${MANOWN}' "" \
+    #    --replace '-g ''${MANGRP}' "" \
+    #    --replace '-o ''${NLSOWN}' "" \
+    #    --replace '-g ''${NLSGRP}' ""
+    #done
+    #set -e
```

Comment:
Is this dead code?

Reply:
It is dead! I kept it around because I wasn't sure if it was right and then forgot about it until it was mass rebuild time :( I will eat that rebuild now